client/data/domains/redirects/use-delete-domain-redirect-mutation.ts

```diff
@@ -0,0 +1,29 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useCallback } from 'react';
+import { DomainsApiError } from 'calypso/lib/domains/types';
+import wp from 'calypso/lib/wp';
+import { domainRedirectsQueryKey } from './domain-redirects-query-key';
+
+export default function useDeleteDomainRedirectMutation(
+	domainName: string,
+	queryOptions: {
+		onSuccess?: () => void;
+		onError?: ( error: DomainsApiError ) => void;
+	}
+) {
+	const queryClient = useQueryClient();
+	const mutation = useMutation( {
+		mutationFn: () => wp.req.post( `/sites/all/domain/${ domainName }/redirects/delete` ),
+		...queryOptions,
+		onSuccess() {
+			queryClient.removeQueries( domainRedirectsQueryKey( domainName ) );
```

Comment:
👋🏻 @lsl I just noticed this. If you're allowing consumers to provide their own `onSuccess` callback you probably want to pass in arguments. RQ provides `data` and `context` there.

```js
onSuccess( ...args ) {
  // …
  queryOptions.onSuccess?.( ...args );
}
```

Ignore my comment if you're doing this on purpose.

Reply:
> Ignore my comment if you're doing this on purpose.

:+1: we just return "success => true" on success so handing down the args isn't helpful.